source/morse.md

```diff
@@ -0,0 +1,336 @@
+	'''
+	def print_sos(nb):
+		s = "." * 3
+		o = "-" * 3
+		stop = "|"
+		print (s+o+s+stop) * nb
+	'''	
+	
+
+Si on voulait l'executer plusieurs fois, on peut imprimer autant de fois print_sos à la fin du fichier. Mais les informaticiens sont flemmards et la machine est là 
+pour nous éviter de refaire la même chose et faire notre travail répétitif et ennuyeux.
+On a besoin de dire à la machine combien de fois on veut imprimer notre SOS.
+On va donc étendre la fonction et lui donner le nb de fois ou on veut imprimer le signal.
+
+Pour qu'on puisse mieux lire les différents SOS et visualiser la fin de la phrase on va lui ajouter une nouvelle ligne. Et souvenez vous python est une super calculatrice capable de
+multiplier du texte 
+
+'''
+	def print_sos(nb):
+		s = "..."
+		o = "---"
+		print (s+o+s+"\n") * nb
+		return
+		
+	print_sos(26)
+'''
+
+#Define a fonction and return
+On a donc une fonction qui prend en entrée le nombre de fois où l'on veut emettre le signal SOS, pour l'instant elle ne se contente que d'afficher.
+Si on veut rendre ce programme encore plus facile à utiliser, imaginons par exemple que nous avons un robot qui transforme le . et le - en sons différents, 
+ou une machine qui allume et éteigne une lampe plus ou moins longtemps.
+On peut vouloir juste que cette fonction emette le signal SOS sans l'imprimer dans le terminal. 
+On va donc demander à la fonction de la retourner et donc on va changer le nom de la fonction de print à emit.
+
+'''
+	def emit_sos(nb):
+		s = "..."
+		o = "---"
+		stop = "|"
+		return (s+o+s+stop) * nb
+		
+	emit_sos(26)
+'''
+
+#Embedding functions 
+Rassurez vous on peut toujours le demander de l'imprimer dans le terminal pour vérifier que cela marche
+'''
+	def emit_sos(nb):
+		s = "..."
+		o = "---"
+		return (s+o+s+"\n") * nb
+		
+	print(emit_sos(26))
+'''
+
+On a donc une fonction utilisable ici on lui a greffé la fonction print à l'exterieur pour la visualiser dans le terminal.
+mais on pourrait faire autre chose par exemple la jouer en son et créer une fonction play au lieu de print.
+'''
+	play(emit_sos(26))
+'''
+
+#Conditions
+Si on a le temps on peut créer une nouvelle fonction qui plutot que d'imprimer des points et des traits 
+emette un son sourt pour le "." et un son long pour "-" et un temps de silence pour découper les mots. 
+
+On va donc d'abord créer une nouvelle fonction qui prend un signal en entrée : un point ou un tiret et renvoie un son.
+
+Pour savoir si c'est un point ou un tiret on va utilise des conditions si(if) alors sinon (else). 
+Quand on a plusieurs conditions on ajoute elif. Ici on a donc 2 conditions: (un . ou un -)
+
+Pour Windows on va utiliser une mini librarie qui permet de faire du son sur l'ordinateur on l'appelle windsound 
+et on la met dans notre programme en l'important avec l'instruction import:
+
+'''
+	
+	import winsound
+	
+	bip_long = winsound.Beep(100,2000)
```

Comment:
I'm not sure if emiting sounds by everyone during the workshops is a good idea. It will be very distracting.